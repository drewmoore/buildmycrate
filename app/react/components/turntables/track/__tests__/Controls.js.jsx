/* globals describe test expect */
import React                   from 'react';
import renderer                from 'react-test-renderer';
import { mount }               from 'enzyme';
import TurntablesTrackControls from '../Controls.js.jsx';
import Fixtures                from '../../../../__tests__/helpers/fixtures.js.es6';
import '../../../../__tests__/helpers/initialize.js.es6';

describe('TurntablesTrackControls', () => {
  const props = {
    playable: true,
    play:     () => {},
    pause:    () => {},
    download: () => {}
  };
  Object.assign(props, Fixtures.tracks.fortySixAndTwo);
  const template = (<TurntablesTrackControls {...props} />);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Rendering Props', () => {
    const listItem  = mount(template);

    test('displays play button if playable', () => {
      expect(listItem.find('[data-hook="track-button-play"]')).toHaveLength(1);
    });

    test('displays pause button if playable', () => {
      expect(listItem.find('[data-hook="track-button-pause"]')).toHaveLength(1);
    });

    test('displays download button if downloadable', () => {
      expect(listItem.find('[data-hook="track-button-download"]')).toHaveLength(1);
    });

    test('displays purchase button if purchaseable', () => {
      expect(listItem.find('[data-hook="track-button-purchase"]')).toHaveLength(1);
    });
  });
});
