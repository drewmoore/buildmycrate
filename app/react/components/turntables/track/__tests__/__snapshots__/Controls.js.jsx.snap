// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurntablesTrackControls DOM Snapshot should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-10 controls"
  >
    <span>
      <img
        alt="Play Button"
        className="track-button clickable"
        data-hook="track-button-play"
        onClick={[Function]}
        src={undefined}
      />
      <img
        alt="Pause Button"
        className="track-button clickable"
        data-hook="track-button-pause"
        onClick={[Function]}
        src={undefined}
      />
    </span>
    <img
      alt="Download Track Button"
      className="track-button clickable"
      data-hook="track-button-download"
      onClick={[Function]}
      src={undefined}
    />
    <a
      className="purchase-link"
      href="https://www.amazon.com/gp/search?ie=UTF8&keywords=Tool"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Purchase Track Button"
        className="track-button clickable"
        data-hook="track-button-purchase"
        src={undefined}
      />
    </a>
  </div>
  <div
    className="col-xs-2 controls text-center"
  >
    <img
      alt="Loading Indicator"
      className="spinner  "
      src={undefined}
    />
  </div>
</div>
`;
